### PR TITLE
UI: Refactor lead_data

### PR DESCRIPTION
**Pull request by deanlee**
Releated issue : https://github.com/commaai/openpilot/issues/2545

### CODE REVIEW

**Review by Torresangl925 — APPROVED** [review]

File: selfdrive/ui/ui.hpp [hunk 1]
```diff
@@ -131,6 +130,13 @@ typedef struct UIScene {
   track_vertices_data track_vertices;
   line_vertices_data lane_line_vertices[4];
   line_vertices_data road_edge_vertices[2];
+
+  // radarState
+  struct LeadData{
+    bool status;
+    float d_rel, v_rel, y_rel;
+    vertex_data vd;
+  } lead[2] = {};
```
Comment (adeebshihadeh) [inline selfdrive/ui/ui.hpp:139]: why create this struct that's just a subset of LeadData from cereal?

**Comment by deanlee** [comment]
@adeebshihadeh  Do you think this would be better?

**Comment by deanlee** [comment]
revert "use cereal::reader". I still think a struct will make the code simpler

**Comment by deanlee** [comment]
merged https://github.com/commaai/openpilot/pull/19705

**Pull request closed without merging**